### DISCUSSION TITLE
Showing hand cursor hovering on links

### DISCUSSION THREAD

**Discussion by nrod** [cefsharp/CefSharp#3473] [Q&A]
Hello CEF# team & community.
I use a browser in a windows WPF application to show some web content. I started using CEF# some time ago when the standard Windows browser component became short for my needs. It was a move in the right direction because this one is really good and has what I wanted. Having said this, thanks everyone who contributed to build this component.

I have, however, a situation. When the mouse is over a standard web page link, like in the first image, the cursor is still a pointer and not a hand like in the second image (_taken from Chrome_). Why is this? Can it be solved?

![link-hover-hand-cursor-missing](https://user-images.githubusercontent.com/3224927/113698327-416ba300-96cc-11eb-9d26-0aa0427190e7.png)

![link-hover-hand-cursor-chrome](https://user-images.githubusercontent.com/3224927/113698816-c060db80-96cc-11eb-9edd-a83e2d934270.png)

Thanks in advance for you help and attention.

**Comment by amaitland** [comment]
Hi,

What version are you using? Does the problem reproduce with https://github.com/cefsharp/CefSharp.MinimalExample

Cursors are working as expected for me. There was a problem with custom cursors a few versions ago, nothing with version `89` that I'm aware of.

**Reply by nrod** [reply]
Thank you @amaitland for your quick answer. A colleague of mine just updated to the last one, version 89.0.170.
I will follow your suggestion and check the situation with the MinimalExample.
I will get back to you once I have more information.

**Reply by nrod** [reply]
@amaitland looks like this situation does not happen in the https://github.com/cefsharp/CefSharp.MinimalExample as analyzed by a colleague of mine. Having reached this conclusion we will investigate what kind of changes have we done that lead to this outcome.
Once more, as soon as I know the cause I will leave a comment here.

**Reply by amaitland** [reply]
We use the https://docs.microsoft.com/en-us/dotnet/api/system.windows.frameworkelement.cursor?view=net-5.0 property to update the cursor in https://github.com/cefsharp/CefSharp/blob/cefsharp/89/CefSharp.Wpf/ChromiumWebBrowser.cs#L1004

I'd suggest reading https://docs.microsoft.com/en-us/dotnet/api/system.windows.frameworkelement.cursor?view=net-5.0#remarks 

Perhaps somewhere you are overriding the cursor?

**Reply by nrod** [reply]
👍🏼 Thank you for your comment and URLs. I will read them and check that possibility. 🤝🏼

**Reply by nrod** [reply]
@amaitland, I have found the problem. Our application does have several places where we change the cursor. Having commented all these situations I could see the normal cursor behaviour in the browser window.
Just for your information, we call some APIs in several places and we have the next bit of code in all those places (_now commented_). Although this kind of cursor management was nice, it is not an important issue.

```
// set the cursor to loading spinner
//Mouse.OverrideCursor = Cursors.Wait;

// call API to get a valid token for this client
ApiResult result = await App.Api.GetAccessToken();

// set the cursor back to arrow
//Mouse.OverrideCursor = Cursors.Arrow;
```

I also had this code there but honestly can't see what it is needed for...

```
//private void Browser_LoadingStateChanged(object sender, CefSharp.LoadingStateChangedEventArgs e) {
//    if (!e.IsLoading) {
//        // set the cursor back to arrow
//        Application.Current.Dispatcher.BeginInvoke(DispatcherPriority.Background,
//            new Action(() => Mouse.OverrideCursor = Cursors.Arrow));
//    }
//}
```

**Thank you so much for the help!**

**Reply by amaitland** [reply]
If you have any suggestions on how we can support this use case then please let me know. We can add a `protected virtual OnCursorChange` to allow overriding the default behaviour.

**Reply by nrod** [reply]
A colleague read somewhere that setting the cursor to a **null** value would keep the behaviour we want. It is being investigated. This way we could still change the cursor to a spinning wheel and set it back to a **default** behaviour. Especifically assigning to `Cursors.Arrow` locks it as an arrow.

```
// set the cursor back to arrow
Mouse.OverrideCursor = null;
```

**Reply by nrod** [reply]
Our final conclusion is that it works fine with this last solution. Changing mouse to **Wait** , do something and then change it back to **null** does not cause any side effect on page links (_on hover_). Once again, thanks for all the help, @amaitland.
